### PR TITLE
Use days of retention when deleting notifications in the nightly job.

### PR DESCRIPTION
**Pull request by servingUpAces**
Update method to delete notifications that are a week old, to look at the days of retention set for the service.

If the service does not have the days of retention set, then use 7 days.
Added a method to get days of retention for a service and notificaiton type

### CODE REVIEW

**Comment by sakisv** [comment]
@govuk-notify-jenkins test this please

File: app/dao/notifications_dao.py [hunk 1]
```diff
@@ -317,21 +319,43 @@ def delete_notifications_created_more_than_a_week_ago_by_type(notification_type)
...
+                                                  Notification.notification_type == notification_type,
+                                                  Notification.service_id.notin_(
+                                                      services_with_data_retention))
+    _delete_letters_from_s3(notification_type=notification_type, query=query)
+    deleted = query.delete(synchronize_session='fetch')
     return deleted
 
 
+def _delete_letters_from_s3(notification_type, query):
+    if notification_type == LETTER_TYPE:
```
Comment (allait) [inline app/dao/notifications_dao.py]: This is minor, but I'd probably move this condition to the task function. Then you won't need the `notification_type` argument and it's clear right away that this doesn't do anything for other notification types.

File: app/dao/notifications_dao.py [hunk 2]
```diff
@@ -317,21 +319,43 @@ def delete_notifications_created_more_than_a_week_ago_by_type(notification_type)
...
+        letters_to_delete_from_s3 = query.all()
+        for letter in letters_to_delete_from_s3:
+            bucket_name = current_app.config['LETTERS_PDF_BUCKET_NAME']
+            sent_at = str(letter.sent_at.date())
```
Comment (allait) [inline app/dao/notifications_dao.py]: Does send date always match the "scheduled" folder date (e.g. could the FTP task fail to run on the date or what if a letter fails to send and ends up with technical failure)?

Comment (servingUpAces) [reply app/dao/notifications_dao.py]: I'm not sure what we'd do for technical failures, it's likely the sent_at date is not set. 
If the letter is not in the sent_at folder, not sure where we'd find it.
The 90 day S3 expiration will kick in.

**Pull request merged**